src/test/java/com/volasoftware/tinder/controllers/FriendControllerTest.java

```diff
@@ -130,6 +139,54 @@ public void testShowFriendInfoWhenFriendIsNotFoundThenExceptionThrown() {
         assertTrue(exception.getMessage().contains(expectedErrorMessage));
     }
 
+    @Test
+    public void rateGivenValidRatingDtoWhenRateFriendThenSuccessfulOperation() {
+        //given
+        RatingDto ratingDto = new RatingDto();
+        ratingDto.setFriendId(1L);
+        ratingDto.setRating(5);
+
+        FriendDto friendDto = new FriendDto();
+        friendDto.setFirstName("Jacob");
+
+        Rating rating = new Rating();
+        rating.setRating(ratingDto.getRating());
+
+        RatingResponseDto ratingResponseDto = new RatingResponseDto(friendDto, rating.getRating());
+        ResponseEntity<?> expectedResponse = ResponseHandler.generateResponse(
+            RatingConstant.RATED_FRIEND,
+            HttpStatus.OK,
+            ratingResponseDto);
+
+        when(ratingService.rateFriend(ratingDto)).thenReturn(ratingResponseDto);
+
+        //when
+        ResponseEntity<?> actualResponse = friendController.rateFriend(ratingDto);
```

Comment:
Im a little bit confused is this a unit or integration test?

Reply:
Test was fixed.